### PR TITLE
DT-1315: ✨ Convert add group to nunjucks

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/controllers/addGroup.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    try {
+      if (!group) {
+        const errors = [{ href: '#group', text: 'Select a group' }]
+        stashStateAndRedirectToIndex(req, res, errors)
+      } else {
+        await saveGroup(res.locals, username, group)
+        res.redirect(staffUrl)
+      }
+    } catch (error) {
+      logError(req.originalUrl, error, serviceUnavailableMessage)
```
Comment (JamieDraperUK) [inline backend/controllers/addGroup.js:48]: Don't change now but we've recently moved away from logging a generic serviceUnavailableMessage message and moved to something like: **The select group page has failed to load.**

Comment (petergphillips) [reply backend/controllers/addGroup.js:48]: okay, can grab the latest PSH and use that instead in a bit

File: backend/controllers/addGroup.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  const post = async (req, res) => {
+    const { username } = req.params
+    const { group } = req.body
+    const staffUrl = `${maintainUrl}/${username}`
+
+    try {
+      if (!group) {
+        const errors = [{ href: '#group', text: 'Select a group' }]
```
Comment (JamieDraperUK) [inline backend/controllers/addGroup.js:41]: Meh, prob doesn't need to be protected by a try catch. Shame you couldn't just do the following before the try catch:

` if(!group) return stashStateAndRedirectToIndex(req, res, errors)`

File: backend/controllers/addGroup.test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        errors: { error: 'some error' },
+        maintainTitle: 'Maintain auth users',
+        maintainUrl: '/maintain-auth-users',
+        groupDropdownValues: [],
+        staff: { name: 'Billy Bob', username: 'BOB' },
+        staffUrl: '/maintain-auth-users/joe',
+      })
+    })
+
+    it('should call error on failure', async () => {
```
Comment (JamieDraperUK) [inline backend/controllers/addGroup.test.js:51]: Is it worth testing that logError is called correctly either here in another test?

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: backend/controllers/addGroup.test.js (hunk 3 above)
Comment (petergphillips) [reply backend/controllers/addGroup.test.js:51]: yes, will add in on next PR

**Pull request merged**